### PR TITLE
Better code for healing reduction

### PR DESCRIPTION
**Pull request by Alex-Turtle**

### CODE REVIEW

File: addons/sourcemod/scripting/vsh/bosses/boss_pyrocar.sp [hunk 1]
```diff
@@ -320,11 +306,15 @@ methodmap CPyroCar < SaxtonHaleBase
...
 			if (!g_hPyrocarHealTimer[victim])
 			{
-				TF2Attrib_SetByDefIndex(victim, ATTRIB_LESSHEALING, 0.5);
-				TF2Attrib_ClearCache(victim);
+				TF2Attrib_SetByDefIndex(GetPlayerWeaponSlot(victim, WeaponSlot_Primary), ATTRIB_LESSHEALING, PYROCAR_HEALINGREDUCTION);
+				TF2Attrib_ClearCache(GetPlayerWeaponSlot(victim, WeaponSlot_Primary));
+				TF2Attrib_SetByDefIndex(GetPlayerWeaponSlot(victim, WeaponSlot_Secondary), ATTRIB_LESSHEALING, PYROCAR_HEALINGREDUCTION);
+				TF2Attrib_ClearCache(GetPlayerWeaponSlot(victim, WeaponSlot_Secondary));
+				TF2Attrib_SetByDefIndex(GetPlayerWeaponSlot(victim, WeaponSlot_Melee), ATTRIB_LESSHEALING, PYROCAR_HEALINGREDUCTION);
+				TF2Attrib_ClearCache(GetPlayerWeaponSlot(victim, WeaponSlot_Melee));
```
Comment (FortyTwoFortyTwo) [inline addons/sourcemod/scripting/vsh/bosses/boss_pyrocar.sp:314]: What's the reason behind this change? I just see it giving more issues than resolving it.
Otherwise if kept, `Destroy` function also need to be updated as it still targets victim instead of weapon.

Comment (Alex-Turtle) [reply addons/sourcemod/scripting/vsh/bosses/boss_pyrocar.sp:314]: The main reason is that the healing reduction is inconsistant and can be avoided by switching weapons.

**Review by FortyTwoFortyTwo — APPROVED** [review]
